crates/router/src/core/payments/flows/session_flow.rs

```diff
@@ -78,6 +82,43 @@ impl Feature<api::Session, types::PaymentsSessionData> for types::PaymentsSessio
     }
 }
 
+fn is_dynamic_fields_required(
+    state: &routes::AppState,
```

Comment:
Can we not take the state here and instead take only the required config. This would help in writing unit tests for this function